### PR TITLE
Temp branch

### PR DESCRIPTION
**Pull request by ddryzhov**
Added User entity, the required DTOs and mapper, AuthenticationController, use data validation in the new DTO classes, Added FieldMatch annotation, RegistrationException using the GlobalExceptionHandler

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/model/User.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@Entity
+@Getter
+@Setter
+@Table(name = "users")
+public class User {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotBlank
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/User.java:23]: Let's use one of annotations NotBlank or Column(nullable = false) 
https://www.baeldung.com/hibernate-notnull-vs-nullable

File: src/main/resources/db/changelog/changes/01-create-user-table.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-user-table
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/01-create-user-table.yaml]:
```suggestion
      id: create-users-table
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's fix some comments

File: src/main/java/mate/academy/bookstore/model/User.java (hunk 1 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/bookstore/model/User.java:23]: Got your choice) 
But, we still need email = unique

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Last comment to consider

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**